script/make_release.py

```diff
@@ -0,0 +1,225 @@
+    return release_post_content
+
+
+def getReleasePost(version, path):
+    release_post_header = f"---\ncategories:\n\t- Release\ntags:\n\t- gh-pages\n\ntitle: {version}\n---\n\n"
+    release_post_headline = "{{page.title}} is live and ready for [download](https://github.com/MaibornWolff/codecharta/releases/tag/{{page.title}}). This version brings the following:\n\n"
+    release_post_content = getLatestChangelogEntry(path)
+    return release_post_header + release_post_headline + release_post_content
+
+
+# Check if we're on project root folder
+if not isRootFolder:
+    print("Please execute this script from the project root folder. Aborting.")
+    quit()
+else:
+    repo = git.Repo(root)
+
+
+# Check if there are any uncommited changes
+if repo.is_dirty():
+    print("Please commit your changes first and/or ignore untracked files in git. Aborting.")
+    quit()
+
+
+# Check if we are on master branch
+if repo.head != "master":
+    print("You can only release on master branch. Aborting.")
+    quit()
+
+
+# Get latest tag
+latest_tag = str(
+    sorted(repo.tags, key=lambda t: t.commit.committed_datetime)[-1])
+print(f"Last version tag in git is {latest_tag}")
+
+
+# Precalc new versions
+versions = latest_tag.split(".")
+
+major = int(versions[0])
+minor = int(versions[1])
+patch = int(versions[2])
+
+new_major_version = f"{major + 1}.0.0"
+new_minor_version = f"{major}.{minor + 1}.0"
+new_patch_version = f"{major}.{minor}.{patch + 1}"
+
+
+# Get release type and version
+questions = [{
+    "type": "list",
+    "name": "version",
+    "message": f"Do you want to release a major version [{new_major_version}], minor version [{new_minor_version}] or a patch [{new_patch_version}]?",
+    "choices": ["Major", "Minor", "Patch"],
+}]
+
+release_type = PyInquirer.prompt(questions)["version"]
+
+if release_type == None:
+    quit()
+if release_type == "Major":
+    new_version = new_major_version
+if release_type == "Minor":
+    new_version = new_minor_version
+if release_type == "Patch":
+    new_version = new_patch_version
+
+
+# Confirm release
+message = f"Do you REALLY want to release {new_version}? WARNING: File changes need to be undone manually or through git when done unintentionally!"
+printMessage = f"Selected {release_type} release. Updating project..."
+confirm(message, printMessage)
+
+
+# bump version in gradle.properties
+gradle_properties = f"{root}/analysis/gradle.properties"
+
+with in_place.InPlace(gradle_properties, encoding="utf-8") as fp:
+    for line in fp:
+        if "currentVersion=" in line:
+            fp.write(
+                f"currentVersion={new_version}\n")
+        else:
+            fp.write(line)
+
+print(f"v{new_version}")
+print("incremented version in ./analysis/gradle.properties")
+
+
+# bump version in package.jsons
+analysis_package = f"{root}/analysis/node-wrapper"
+analysis_package_json = f"{analysis_package}/package.json"
+analysis_package_lock_json = f"{analysis_package}/package-lock.json"
```

Comment:
I could not figure out what's done with the package-lock. Why would that change?

Reply:
Because we bump the version in the package.json and package-lock.json using npm version

```
# bump version in package.jsons
npm --prefix ./analysis/node-wrapper --no-git-tag-version version $NEW_VERSION
echo "incremented version in ./analysis/node-wrapper/package.json + locks"

npm --prefix ./visualization --no-git-tag-version version $NEW_VERSION
echo "incremented version in ./visualization/package.json + locks"
```